### PR TITLE
103 add-streaminghandler-and-settingshandler

### LINKED ISSUE #103: Add `streamingHandler` and `settingsHandler`
Add missing handlers to handle grpc api call

### PR DESCRIPTION
**Pull request by GravityDarkLab**
## Add

`StreamHandler` to fetch and get streams and Thumbnails

## Resolves

closes #103

### CODE REVIEW

File: lib/base/networking/api/handler/stream_handler.dart [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    );
+  }
+
+  /// Fetches the thumbnail stream.
+  ///
+  /// This method sends a `getThumbsLive` gRPC call to fetch the thumbnail stream.
+  ///
+  /// Takes [streamId] as a parameter.
+  /// Returns a [String] instance that represents the thumbnail stream.
+  Future<String> fetchThumbnailStreams(Int64? streamId) async {
```
Comment (carlobortolan) [inline lib/base/networking/api/handler/stream_handler.dart:72]: Is there a reason why the streamID is marked as optional? ```Int64?```

File: lib/base/networking/api/handler/stream_handler.dart [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    );
+  }
+
+  /// Fetches the thumbnail VOD.
+  ///
+  /// This method sends a `getThumbsVOD` gRPC call to fetch the thumbnail VOD.
+  ///
+  /// Takes [streamId] as a parameter.
+  /// Returns a [String] instance that represents the thumbnail VOD.
+  Future<String> fetchThumbnailVOD(Int64? streamId) async {
```
Comment (carlobortolan) [inline lib/base/networking/api/handler/stream_handler.dart:90]: Is there a reason why the streamID is marked as optional? ```Int64?```

File: lib/base/networking/api/handler/stream_handler.dart [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  }
+
+  /// Fetches the stream of the course.
+  ///
+  /// This method sends a `getStream` gRPC call to fetch the stream of the course.
+  ///
+  /// Takes [streamId] as a parameter.
+  ///
+  /// Returns a [Stream] instance that represents the stream of the course.
+  Future<Stream> fetchStream(Int64? streamId) async {
```
Comment (carlobortolan) [inline lib/base/networking/api/handler/stream_handler.dart:38]: Is there a reason why the streamID is marked as optional? ```Int64?```

Comment (GravityDarkLab) [reply lib/base/networking/api/handler/stream_handler.dart]: Fix ✅

File: lib/base/networking/api/handler/stream_handler.dart (hunk 2 above)
Comment (GravityDarkLab) [reply lib/base/networking/api/handler/stream_handler.dart]: Fix ✅

File: lib/base/networking/api/handler/stream_handler.dart (hunk 1 above)
Comment (GravityDarkLab) [reply lib/base/networking/api/handler/stream_handler.dart]: Fix ✅

**Review by carlobortolan — DISMISSED** [review]
Looks good to me :+1:

**Review by carlobortolan — APPROVED** [review]

**Pull request merged**